compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1045,6 +1045,9 @@ pub struct GlobalCtxt<'tcx> {
     output_filenames: Arc<OutputFilenames>,
 
     pub main_def: Option<MainDefinition>,
+
+    pub(super) vtables_cache:
+        Lock<FxHashMap<(Ty<'tcx>, Option<ty::PolyExistentialTraitRef<'tcx>>), AllocId>>,
```

Comment:
Hmm, any reason this is a mutable cache in the global context, instead of a query? It looks like the kind of thing we replaced with queries long ago.

(Maybe we should find a way to ban `!Freeze` types in `GlobalCtxt`, outside of incremental/query state?)

cc @oli-obk @wesleywiser

Reply:
I actually gave it a try as a query locally, but without succeed. The compilation complained about keys becoming bigger, something like that, i can't remember the exact error content.

Reply:
Ah, that's because that key type is a bit too "spread out". Sadly you can't just use `TraitRef<'tcx>`, without passing some dummy trait for the `DefId`.

You could "just" update the expected size, but someone would need to check off on that (and I don't know who) - the problem there is it can affect performance even if the new query is completely unused.

Reply:
Having a cache without any dependency tracking looks like a bug to me.